classes/Address.php

```diff
@@ -232,6 +236,11 @@ public function delete()
         if (!$this->isUsed()) {
             $this->deleteCartAddress();
 
+            // Update the cache
+            if (isset(self::$addressExists[$this->id])) {
+                self::$addressExists[$this->id] = false;
+            }
```

Comment:
```suggestion
            if (isset(static::$addressExists[$this->id])) {
                static::$addressExists[$this->id] = false;
            }
```

Avoid the usage of `self` since this class can be extended and the atribute is protected

Reply:
Fixed